troubleshooting/tutorials/remote-core-dump/README.md

```diff
@@ -0,0 +1,309 @@
+
+
+- [Step 1: Debug Build of Fluent Bit](#step-1-debug-build-of-fluent-bit)
+- [Step 2: Modifying your deployment to capture a core file](#step-2-modifying-your-deployment-to-capture-a-core-file)
+    - [Setup for Local Dev Machine or Instance](#setup-for-local-dev-machine-or-instance)
+    - [Setup for ECS EC2](#setup-for-ecs-ec2)
+    - [Setup for ECS Fargate](#setup-for-ecs-fargate)
+    - [Setup for EKS EC2](#setup-for-eks-ec2)
+- [Step 3: Using GDB with a live Fluent Bit](#step-3-using-gdb-with-a-live-fluent-bit)
+- [Step 4: Using GDB with a core file (crashed Fluent Bit)](#step-4-using-gdb-with-a-core-file-crashed-fluent-bit)
+
+
+## Step 1: Debug Build of Fluent Bit
+
+Clone the AWS for Fluent Bit source code, and then edit the `Dockerfile.debug` so that the valgrind entrypoint is removed and the core file debug entrypoint is un-commented. The final lines of the Dockerfile should be:
+
+```
+RUN mkdir /cores && chmod 777 /cores
+CMD /fluent-bit/bin/fluent-bit -c /fluent-bit/etc/fluent-bit.conf
+```
+
+When Fluent Bit crashes, a core file will be dumped to the `/cores` directory because the default kernel core pattern in the image is: `/cores/core.%e.%p`.
+
+There are couple of things to note about the `Dockerfile.debug` for the core file debugging use case:
+- The Fluent Bit upstream base version is specified with `ENV FLB_VERSION`
+- Fluent Bit is compiled with CMake flag `-DFLB_DEBUG=On`
+- `gdb` is installed in the final stage of the Docker build.
+
+When you clone AWS for Fluent Bit, you will automatically get the latest Dockerfile for our latest release on the mainline branch. To create a debug build of a different version, either check out the tag for that version, or modify the `ENV FLB_VERSION` at the top of the Dockerfile to install the desired Fluent Bit base version.
+
+Once you are ready, build the debug image:
+
+```
+make debug
+```
+
+And then push this image to a container image repository such as Amazon ECR so that you can use it in your deployment in the next step. 
+
+
+## Step 2: Modifying your deployment to capture a core file
+
+Follow the version of this step that fits your deployment mode. 
+
+- [Setup for Local Dev Machine or Instance](#setup-for-local-dev-machine-or-instance)
+- [Setup for ECS EC2](#setup-for-ecs-ec2)
+- [Setup for ECS Fargate](#setup-for-ecs-fargate)
+- [Setup for EKS EC2](#setup-for-eks-ec2)
+
+### Setup for Local Dev Machine or Instance
+
+Simply run the debug build of Fluent Bit with ulimit unlimited and with the `/cores` directory mounted onto your host:
+```
+docker run --ulimit core=-1 -v /somehostpath:/cores -v $(pwd):/fluent-bit/etc amazon/aws-for-fluent-bit:debug
+```
```

Comment:
I might suggest that this command also include capabilities to ensure gdb can create a core-dump if the process is still active - `--cap-add=SYS_PTRACE --security-opt seccomp=unconfined`

Reply:
Thanks, added a note on this